olp-cpp-sdk-core/include/olp/core/client/OlpClientSettings.h

```diff
@@ -186,7 +183,8 @@ struct OlpClientSettings {
    * @brief The network instance to be used to internally operate with OLP
    * services.
    */
-  std::shared_ptr<http::Network> network_request_handler = nullptr;
+  std::shared_ptr<http::Network> network_request_handler =
+      http::CreateDefaultNetwork();
```

Comment:
This is not good. Means everybody creates a new Network instance before it can share and already existing one.
Means:

```cpp
// Create settings to pass to clients -> 1. Network instance created 
auto settings1 = std::make_shared<olp::client::OlpClientSettings>();
// Setup client settings
...
// Create second settings to pass to other client, BAD - 2. Network instance created
auto settings2 = std::make_shared<olp::client::OlpClientSettings>();
settings2->network_request_handler = settings1->network_request_handler;
```

Move back to nullptr initialization.